packages/rrweb-snapshot/test/rebuild.test.ts

```diff
@@ -81,47 +85,90 @@ describe('rebuild', function () {
   describe('add hover class to hover selector related rules', function () {
     it('will do nothing to css text without :hover', () => {
       const cssText = 'body { color: white }';
-      expect(addHoverClass(cssText, cache)).toEqual(cssText);
+      expect(adaptCssForReplay(cssText, cache)).toEqual(cssText);
     });
 
     it('can add hover class to css text', () => {
       const cssText = '.a:hover { color: white }';
-      expect(addHoverClass(cssText, cache)).toEqual(
+      expect(adaptCssForReplay(cssText, cache)).toEqual(
         '.a:hover, .a.\\:hover { color: white }',
       );
     });
 
+    it('can correctly add hover when in middle of selector', () => {
+      const cssText = 'ul li a:hover img { color: white }';
+      expect(adaptCssForReplay(cssText, cache)).toEqual(
+        'ul li a:hover img, ul li a.\\:hover img { color: white }',
+      );
+    });
+
+    it('can correctly add hover on multiline selector', () => {
+      const cssText = `ul li.specified a:hover img,
+ul li.multiline
+b:hover
+img,
+ul li.specified c:hover img {
+  color: white
+}`;
+      expect(adaptCssForReplay(cssText, cache)).toEqual(
+        `ul li.specified a:hover img, ul li.specified a.\\:hover img,
+ul li.multiline
+b:hover
+img, ul li.multiline
+b.\\:hover
+img,
+ul li.specified c:hover img, ul li.specified c.\\:hover img {
+  color: white
+}`,
```

Comment:
I think you can add whitespace here to make the tests a little less messy, unless I'm missing something it should be fine.
```suggestion
         ul li.multiline
         b:hover
         img,
         ul li.specified c:hover img {
           color: white
         }`;
      expect(adaptCssForReplay(cssText, cache)).toEqual(
        `ul li.specified a:hover img, ul li.specified a.\\:hover img,
         ul li.multiline
         b:hover
         img, ul li.multiline
         b.\\:hover
         img,
         ul li.specified c:hover img, ul li.specified c.\\:hover img {
           color: white
         }`,
```